Adding campaign republish behavior docs (#561)
mautic/user-documentation: docs/campaigns/managing_campaigns.rst

The **Contacts** tab displays a grid view of all the Contacts that you have added to your Campaign.

The **Recent Activity** panel on the right displays the recent activities that have taken place in the Campaign.

.. vale off

.. _campaign republish behavior:

Campaign republish behavior
***************************

.. vale on

> When you Unpublish and then republish a Campaign, Mautic provides control over how scheduled events with relative delays - such as "Send Email 5 days after joining" - should behave. This feature gives you flexibility in managing Campaign timing based on your specific use case.

Reviewer: This functionality is now called 'activate' and 'deactivate' so it's going to need updating throughout this PR.

**Example scenario:**

- Campaign trigger date: January 1
- Event delay: 10 days
- Original calculated event date: January 11
- Campaign unpublished: January 5 (after 4 days published)
- Campaign republished: January 10 (after 5 days unpublished)

**Result:** the event is rescheduled to execute on January 16. The 4 days while published (January 1-5) count toward the 10-day delay, leaving 6 more days needed after republishing (January 10 + 6 days = January 16).

**When to use:** this option is ideal when you want precise control over the actual time Contacts spend in an active Campaign state. Use this for compliance scenarios, trial periods, or when you need to pause Campaigns without affecting the intended engagement timeline.

Viewing last publish date
=========================

On the Campaign details page, you can see the **Last Publish Date** which indicates when the Campaign was most recently published. This date is used as the reference point for the "Restart on republish" option to recalculate scheduled event timings.

Publish and Unpublish Campaigns
======================================

> When you publish or Unpublish a Campaign, Mautic displays a confirmation message that shows the current republish behavior setting. This helps you understand what happens to scheduled events before you confirm the action.

Reviewer: Let's add a screenshot here.

Author: Oh man. The wording is not consistent in the UI. Here are the screenshots but I think we'll have to update the wording in the product too, right?
<img width="566" height="413" alt="Screenshot 2025-12-09 at 15 52 08" src="https://github.com/user-attachments/assets/8786c60f-12a7-4b60-9e38-f7fc5fcbb227" />
<img width="441" height="230" alt="Screenshot 2025-12-09 at 15 51 55" src="https://github.com/user-attachments/assets/83640e29-62fa-4063-9ac1-39fa6a0180ad" />

Reviewer: Yes, we will indeed. cc @andersonjeccel for visibility.

Author: What would be the re-wording for republish? Reactivate?

Reviewer: I believe it should now be:

Active / Activate / Reactivate
Inactive / Deactivate 

Referencing this PR: https://github.com/mautic/mautic/pull/14239 and this PR: https://github.com/mautic/mautic/pull/13500

Author: So the new setting label should be called "Campaign Reactivate Behaviour"?

Reviewer: Resolving this as it's been addressed, from what I saw in the PR.

The **Contacts** tab displays a grid view of all the Contacts that you have added to your Campaign.

The **Recent Activity** panel on the right displays the recent activities that have taken place in the Campaign.

.. vale off

.. _campaign reactivate behavior:

Campaign reactivation behaviour

> ********************************

Reviewer 2: Let's remove excess punctuation for title here and some others in the following review.

```suggestion
Campaign reactivation behaviour
*******************************
```

.. vale on

> When you deactivate and then reactivate a Campaign, Mautic provides control over how scheduled events with relative delays - such as "Send Email 5 days after joining" - should behave. This feature gives you flexibility in managing Campaign timing based on your specific use case.

Reviewer: ```suggestion
When you deactivate and then reactivate a Campaign, Mautic provides control over how scheduled events with relative delays - such as 'Send Email 5 days after joining' - should behave. This feature gives you flexibility in managing Campaign timing based on your specific use case.
```
This is not speech, so I'd use single quotes here.

.. note::

>     This setting only affects events that use relative delays (interval-based scheduling). Events with absolute dates aren't affected by this setting.

Reviewer 2: We can use dash here instead of parantheses.

```suggestion
    This setting only affects events that use relative delays - interval-based scheduling. Events with absolute dates aren't affected by this setting.
```

Reviewer 2: Or would the suggestion below be better if it's not out of the context?

```suggestion
    This setting only affects events that use interval-based scheduling for relative delays. Events with absolute dates aren't affected by this setting.
```

Configuring reactivation behaviour

> ===================================

Reviewer 2: ```suggestion
Configuring reactivation behaviour
==================================
```

You can configure the reactivation behaviour at two levels:

1. **Global default** - Set in Configuration > Campaign Settings > Campaign Reactivation Behaviour. This applies to all Campaigns unless overridden.
2. **Per Campaign** - Set when creating or editing a Campaign. This overrides the global default for that specific Campaign.

Reactivation behaviour options

> ===============================

Reviewer 2: ```suggestion
Reactivation behaviour options
==============================
```

There are three options available for how scheduled events should behave after reactivation:

Count delay regardless of activation state

> -------------------------------------------

Reviewer 2: ```suggestion
Count delay regardless of activation state
------------------------------------------
```

> This is the default behaviour. The original trigger date is used and inactive time doesn't affect scheduling.

Reviewer: ```suggestion
This is the default behaviour. Mautic uses the original trigger date, and inactive time doesn't affect scheduling.
```

**Example scenario:**

- Campaign trigger date: January 1
- Event delay: 10 days
- Calculated event date: January 11
- Campaign deactivated: January 5
- Campaign reactivated: January 7

**Result:** the event still executes on January 11, as originally scheduled.

**When to use:** this option maintains the original scheduled timing, treating the Campaign's activation state as irrelevant to the delay calculation. Use this when you want consistency with the original schedule, or when temporarily deactivating a Campaign shouldn't affect when events execute.

Restart on reactivation

> ---------------------

Reviewer 2: The dashes here should be as long as the title.

```suggestion
Restart on reactivation
-----------------------
```

The delay counter resets completely when you reactivate the Campaign.

**Example scenario:**

- Campaign trigger date: January 1
- Event delay: 10 days
- Original calculated event date: January 11
- Campaign deactivated: January 5
- Campaign reactivated: January 7

> **Result:** the event is rescheduled to execute 10 days after reactivation, on January 17.

Reviewer: ```suggestion
**Result:** Mautic reschedules the event to execute 10 days after reactivation, on January 17.
```

**When to use:** this option is useful when you want to ensure all Contacts receive the full intended delay after any Campaign changes. For example, if you deactivate a Campaign to make significant updates and want everyone to experience the complete updated workflow timing.

Count delay only while active

> ------------------------------

Reviewer 2: ```suggestion
Count delay only while active
-----------------------------
```

Events only count days when the Campaign is active. Inactive periods don't count toward the delay.

**Example scenario:**

- Campaign trigger date: January 1
- Event delay: 10 days
- Original calculated event date: January 11
- Campaign deactivated: January 5 (after 4 days active)
- Campaign reactivated: January 10 (after 5 days inactive)

> **Result:** the event is rescheduled to execute on January 16. The 4 days while active (January 1-5) count toward the 10-day delay, leaving 6 more days needed after reactivation (January 10 + 6 days = January 16).

Reviewer: ```suggestion
**Result:** Mautic reschedules the event to execute on January 16. The 4 days while active (January 1-5) count toward the 10-day delay, leaving 6 more days needed after reactivation (January 10 + 6 days = January 16).
```

**When to use:** this option is ideal when you want precise control over the actual time Contacts spend in an active Campaign state. Use this for compliance scenarios, trial periods, or when you need to pause Campaigns without affecting the intended engagement timeline.

Viewing last activation date

> =============================

Reviewer 2: ```suggestion
Viewing last activation date
============================
```

> On the Campaign details page, you can see the **Last Publish Date** which indicates when the Campaign was most recently activated. This date is used as the reference point for the "Restart on reactivate" option to recalculate scheduled event timings.

Reviewer: ```suggestion
On the Campaign details page, you can see the **Last Publish Date** which indicates when the Campaign was most recently activated. This date is used as the reference point for the 'Restart on reactivate' option to recalculate scheduled event timings.
```

Activate and deactivate Campaigns

> ==================================

Reviewer 2: ```suggestion
Activate and deactivate Campaigns
=================================
```

When you activate or deactivate a Campaign, Mautic displays a confirmation message that shows the current reactivation behaviour setting. This helps you understand what happens to scheduled events before you confirm the action.

> For example: "All scheduled events execute according to the reactivation behaviour setting. Currently set to: Count delay regardless of activation state"

Reviewer: ```suggestion
For example: 'All scheduled events execute according to the reactivation behaviour setting. Currently set to: Count delay regardless of activation state'
```
